Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyTreeView.cxx

```diff
@@ -1346,7 +1358,26 @@ void qMRMLSubjectHierarchyTreeView::toggleSubjectHierarchyItemVisibility(vtkIdTy
     d->SubjectHierarchyNode->GetScene()->StartState(vtkMRMLScene::BatchProcessState);
     }
 
-  int visible = (ownerPlugin->getDisplayVisibility(itemID) > 0 ? 0 : 1);
+  int visible;
+  switch (visibilityAction)
+    {
+    case 0:
+      {
+      visible = 0;
+      break;
+      }
+    case 1:
+      {
+      visible = 1;
+      break;
+      }
+    case 2:
+      {
+      visible = (ownerPlugin->getDisplayVisibility(itemID) > 0 ? 0 : 1);
+      }
```

Comment:
Probably want to consider the usage of an official enum for clarity about what 0, 1, and 2 mean here:

https://github.com/Slicer/Slicer/blob/ed6fd97d5d098d384fe4e09c6fac2cd7c7e4fd83/Libs/MRML/Core/vtkMRMLSliceCompositeNode.h#L138-L145
https://github.com/Slicer/Slicer/blob/ed6fd97d5d098d384fe4e09c6fac2cd7c7e4fd83/Libs/MRML/Widgets/qMRMLSliceControllerWidget.cxx#L1079-L1090

Reply:
We could use QItemSelectionModel::SelectionFlag enums:
        Select         = 0x0002,
        Deselect       = 0x0004,
        Toggle         = 0x0008,

Reply:
That sounds like a good idea. Will the meaning Deselect=Hide and Select=Show be clear in this context? Or would it be better to define a new enum?